modules.d/90dmsquash-live-autooverlay/create-overlay.sh

```diff
@@ -55,45 +124,69 @@ gatherData() {
     read -r readonly < ${rootDeviceSysfsPath}/ro
     # shellcheck disable=SC2086
     if [ "$partition" != "1" ] || [ "$readonly" != "0" ]; then
-        info "Skipping overlay creation: unpartitioned or read-only media detected"
+        info "Skipping overlay creation: unpartitioned or read-only media detected."
         exit 0
     fi
     # shellcheck disable=SC2046
     # shellcheck disable=SC2086
-    fullDriveSysfsPath=$(readlink -f ${rootDeviceSysfsPath}/..)
-    blockDevice=/dev/${fullDriveSysfsPath##*/}
-    currentPartitionCount=$(grep --count -E "${blockDevice#/dev/}[0-9]+" /proc/partitions)
-
-    # shellcheck disable=SC2086
-    freeSpaceStart=$(parted --script ${blockDevice} unit % print free \
-        | awk -v x=${currentPartitionCount} '$1 == x {getline; print $1}')
-    if [ -z "$freeSpaceStart" ]; then
-        info "Skipping overlay creation: there is no free space after the last partition"
-        exit 0
+    if [ "$partition" ]; then
+        fullDriveSysfsPath=$(readlink -nf ${rootDeviceSysfsPath}/..)
+    else
+        fullDriveSysfsPath=${rootDeviceSysfsPath}
     fi
-    partitionStart=$((${freeSpaceStart%.*} + 1))
-    if [ $partitionStart -eq 100 ]; then
-        info "Skipping overlay creation: there is not enough free space after the last partition"
+    blockDevice=${fullDriveSysfsPath##*/}
+
+    IFS='
+'
+    set -- $(runParted /dev/"${blockDevice}" --fix -m unit B print free)
+    currentPartitionCount=$(eval echo \$"$(($# - 1))")
+    currentPartitionCount=${currentPartitionCount%%:*}
+    IFS=' :'
+    set -- $(eval echo \$"$#")
+    IFS=' 	
+'
+    freeSpaceStart=${2%B}
+    freeSpaceEnd=${3%B}
+    freeSpaceAvailable=${4%B}
+    if [ "$freeSpaceAvailable" -lt "$((1<<27))" ]; then
```

Comment:
It was 1% of disc in the earlier, undistributed version.  This may be too small on very small devices. 1% of an 8 GiB disc would be about 86 MiB or 172 MiB on a 16 GiB disc (currently common sizes).  So 128 MiB, the average of those  seemed like a convenient number.

Even this may be too small. I experience noticeable sluggishness  on such discs when the there is only a few percent of free space available.  After some more testing and experience, this number should probably be adjusted.

Even larger discs may need more free space.  What information or experience do we have for the 1 % value?

We should probably raise the minimum 5 or 10 times for a more satisfying experience and fewer complaints to deal with.  10 % seems like a number that has been recommended before.  Maybe that was the intent in the first place and I misread the code.